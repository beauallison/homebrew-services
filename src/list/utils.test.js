const fixtures = require('../../fixtures');
const { convertToMap } = require('./utils');

describe('list/convertToMap()', () => {
  it('should convert a list of services to a map', () =>
    expect(convertToMap(fixtures.get('list'))).toMatchSnapshot());

  it('should handle empty results', () =>
    expect(convertToMap('')).toEqual(new Map()));

  it('should handle undefined inputs', () =>
    expect(convertToMap()).toEqual(new Map()));
});
