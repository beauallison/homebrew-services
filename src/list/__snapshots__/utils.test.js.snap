// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`list/convertToMap() should convert a list of services to a map 1`] = `
Map {
  "cassandra" => "stopped",
  "chronograf" => "stopped",
  "elasticsearch" => "stopped",
  "influxdb" => "stopped",
  "kafka" => "stopped",
  "kapacitor" => "stopped",
  "mongodb" => "stopped",
  "rabbitmq" => "stopped",
  "redis" => "started",
  "zookeeper" => "stopped",
}
`;
